pallets/order-book/src/lib.rs

```diff
@@ -175,6 +175,18 @@ pub mod pallet {
 			Error = DispatchError,
 		>;
 
+		/// Type for a market price feeder
+		type FeederId: Parameter + Member + Ord + MaxEncodedLen;
+
+		/// Account used to feed with market prices
+		type MarketFeederId: Get<Self::FeederId>;
+
+		/// Identification for a market price
+		type Pair: From<(Self::AssetCurrencyId, Self::AssetCurrencyId)>;
+
+		/// A way to obtain prices for market pairs
+		type PriceProvider: ValueProvider<Self::FeederId, Self::Pair, Value = Self::SellRatio>;
+
```

Comment:
Right now, only one feeder (`MarketFeederId`) can feed. `PriceProvider` will be `pallet-oracle-feed` (which does not have validation & aggregation). 

In order to have the aggregation and validation features from `pallet-oracle-collection`, I was thinking of making `pallet-oracle-collection` instantiable. One instance will be used by `pallet-loans` where `CollectionId` is `PoolId`. And another instance will be used by `pallet-order-book`, where `CollectionId` can be something different than a pool id, like a `MarketId`. 

That way, we configure `pallet-order-book` with a `PriceProvider` which is a `pallet-oracle-collection` where exists a collection of market oracle values, already aggregated and validated. `ChangeGuard` of this instance of `pallet-oracle-collection` can be something different than `pool-system`, which always allows the change for this kind of admin.

Just a though for a next step improvement, but let me your thoughts @hieronx @mustermeiszer

Reply:
Sounds like a good idea for the next iteration. If you do not want to introduce the extrinsic and storage for setting the feeder dynamiclaly in this iteration, I would also be fine with using another instance of `collection` already in this run.

Reply:
I'll evaluate if I have time to use the collection. I probably have a new PR over this if I have time to introduce it before the deadline.